lib/index.js

```diff
@@ -224,10 +226,86 @@ function getPolyfillString(options) {
 	;
 }
 
+function getDetectString(options) {
+	const features = options ? options.features : undefined;
+	let callback;
+	if (options && typeof options.callback === 'string') {
+		callback = options.callback;
+	}
+	const unknown = [];
+
+	return Promise.resolve(features)
+		.then(resolveAliases)
+		.then(features => {
+			const featureNames = Object.keys(features);
+
+			function createDetectFor (name) {
+				const graph = tsort();
+				const featureDetects = new Map();
+				const createDetects = co.wrap(function* r (name) {
+					const polyfill = yield sourceslib.getPolyfill(name);
+
+					if (!polyfill) {
+						unknown.push(name);
+					} else {
+						graph.add(name);
+
+						if (polyfill.dependencies) {
+							for (const depName of polyfill.dependencies) {
+								if (!unknown.includes(depName)) {
+									graph.add(depName, name);
+									yield r(depName);
+								}
+							};
+						}
+						featureDetects.set(name, stripIndents`(
+							${polyfill.detectSource || false}
+							)`);
+					}
+				});
+
+				return createDetects(name).then(() => {
+					const detects = graph.sort().filter(name => !unknown.includes(name)).map(name => {
+						return featureDetects.get(name);
+					}).join(' && ');
+					return `if (!(${detects})) { featuresToPolyfill.push('${name}')}`;
+				});
+			}
+
+			return Promise.all(featureNames.map(createDetectFor));
+		})
+		.then(detects => {
+
+			// Outer closure hides private features from global scope
+			// Invoke the closure, binding `this` to window (in a browser), self (in a web worker), or global (in Node/IOjs)
+			let featureDetectBundle = stripIndents`(function(undefined) {
+				var featuresToPolyfill = [];
+				${detects.join('')}
+				${callback ? `${callback}(featuresToPolyfill);` : ''}
+				return featuresToPolyfill;
+			}).call('object' === typeof window && window || 'object' === typeof self && self || 'object' === typeof global && global || {});`;
```

Comment:
A template string inside another template string strikes me as confusing code.  It might seem weird to use a library over the built in feature, but we already have handlebars as a dependency of the project,and id be tempted to use it here like i've done in the RUM template.  This then also avoids strings of client side code being interspersed with servr side code in the same file, and means you can avoid adding a dependency (`common-tags`) to the project.